Makefile

```diff
@@ -37,92 +36,43 @@ CROSS_COMPILE_32 ?= arm-none-linux-gnueabihf-
-	$(Q)$(MAKE) -C $(K3IMGGEN_DIR) SOC=$(K3IMGGEN_SOC) SOC_TYPE=$(SECURITY_TYPE) CONFIG=evm CROSS_COMPILE=$(CROSS_COMPILE_32) \
-			SYSFW_DIR=$(FW_DIR)/ti-sysfw O=$(O)/k3-img-gen TI_SECURE_DEV_PKG=$(SECDEV_DIR) mrproper
-	$(Q)$(MAKE) -C $(K3IMGGEN_DIR) SOC=$(K3IMGGEN_SOC) SOC_TYPE=$(SECURITY_TYPE) CONFIG=evm CROSS_COMPILE=$(CROSS_COMPILE_32) \
-			SYSFW_DIR=$(FW_DIR)/ti-sysfw O=$(O)/k3-img-gen TI_SECURE_DEV_PKG=$(SECDEV_DIR) SYFW_DIR=$(FW_DIR)/ti-sysfw
-	$(Q)cp -v $(K3IMGGEN_DIR)/sysfw.itb $(D)
-
 tfa: $(O) $(I)
 	$(Q)$(MAKE) -C $(TFA_DIR) BUILD_BASE=$(O)/arm-trusted-firmware CROSS_COMPILE=$(CROSS_COMPILE_64) ARCH=aarch64 PLAT=k3 TARGET_BOARD=$(TFA_BOARD) $(TFA_EXTRA_ARGS) SPD=opteed all
 	$(Q)cp -v $(O)/arm-trusted-firmware/k3/$(TFA_BOARD)/release/bl31.bin $(I)
-ifneq ($(SECURITY_TYPE),gp)
-	$(SECDEV_DIR)/scripts/secure-binary-image.sh $(O)/arm-trusted-firmware/k3/$(TFA_BOARD)/release/bl31.bin $(O)/arm-trusted-firmware/k3/$(TFA_BOARD)/release/bl31.bin$(HS_EXT)
-	$(Q)cp -v $(O)/arm-trusted-firmware/k3/$(TFA_BOARD)/release/bl31.bin$(HS_EXT) $(I)
-endif
 
 optee: $(O) $(I)
 	$(Q)$(MAKE) -C $(OPTEE_DIR) O=$(O)/optee CROSS_COMPILE=$(CROSS_COMPILE_32) CROSS_COMPILE64=$(CROSS_COMPILE_64) PLATFORM=$(OPTEE_PLATFORM) $(OPTEE_EXTRA_ARGS) CFG_TEE_CORE_LOG_LEVEL=2 CFG_TEE_CORE_DEBUG=y CFG_ARM64_core=y all
-	$(Q)cp -v $(O)/optee/core/tee-pager_v2.bin $(I)/
-ifneq ($(SECURITY_TYPE),gp)
-	$(SECDEV_DIR)/scripts/secure-binary-image.sh $(O)/optee/core/tee-pager_v2.bin $(O)/optee/core/tee-pager_v2.bin$(HS_EXT)
-	$(Q)cp -v $(O)/optee/core/tee-pager_v2.bin$(HS_EXT) $(I)/
-endif
-
-dm: $(I)
-ifneq ($(DM_COMBINED_WITH_TIFS),1)
-	$(Q)cp -v $(FW_DIR)/ti-dm/$(DM_SOC_NAME)/ipc_echo_testb_mcu1_0_release_strip.xer5f $(I)
-ifneq ($(SECURITY_TYPE),gp)
-	$(SECDEV_DIR)/scripts/secure-binary-image.sh $(I)/ipc_echo_testb_mcu1_0_release_strip.xer5f $(I)/ipc_echo_testb_mcu1_0_release_strip.xer5f$(HS_EXT)
-endif
-endif
-
-u_boot_r5: $(O) $(I)
-	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_32) ARCH=arm TI_SECURE_DEV_PKG=$(SECDEV_DIR) O=$(O)/u-boot/r5 $(UBOOT_ARMV7_DEFCONFIG)
-	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_32) ARCH=arm TI_SECURE_DEV_PKG=$(SECDEV_DIR) O=$(O)/u-boot/r5
-ifeq ($(MULTICERTIFICATE_BOOT_CAPABLE),1)
-	$(Q)cp -v $(O)/u-boot/r5/spl/u-boot-spl.bin $(I)
-else
-	$(Q)cp -v $(O)/u-boot/r5/tiboot3.bin $(D)
-endif
-
-u_boot_armv8: $(O) $(D) optee tfa dm
-	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_64) ARCH=arm TI_SECURE_DEV_PKG=$(SECDEV_DIR) O=$(O)/u-boot/armv8 $(UBOOT_ARMV8_DEFCONFIG)
-	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_64) ARCH=arm TI_SECURE_DEV_PKG=$(SECDEV_DIR) O=$(O)/u-boot/armv8 \
-					ATF=$(I)/bl31.bin$(HS_EXT) \
-				  TEE=$(I)/tee-pager_v2.bin$(HS_EXT) \
-				  $(DMCONF)
-	$(Q)cp -v $(O)/u-boot/armv8/tispl.bin$(HS_UBOOTEXT) $(D)/tispl.bin
-	$(Q)cp -v $(O)/u-boot/armv8/u-boot.img$(HS_UBOOTEXT) $(D)/u-boot.img
+	$(Q)cp -v $(O)/optee/core/tee-pager_v2.bin $(I)
+
+u_boot_r5: $(O) $(D)
```

Comment:
For legacy boot (AM65 and J721e) you also need to deploy sysfw.itb from the u-boot build.